### ISSUE TITLE
RET is mu4e-scroll-up and shr-browse-url

### ISSUE THREAD

**Issue by bradyt** [djcb/mu#1216]
## Expected or desired behavior

Hold `RET` key down to scroll buffer up.

## Actual behavior

When I hold `RET` down for `mu4e-scroll-up`, buffer scrolls up until point encounters a line that starts with a link, then browses to it multiple times as a result of key repeat rate.

## Steps to reproduce

`(setq mu4e-html2text-command 'mu4e-shr2text)`

If point is on a link, `RET` is now bound to `shr-browse-url`. For more annoying behavior, try the following.

1. In OS settings, I have key repeat set to fast, and delay until repeat set to short.
2. Find an html email that renders with some line starting with a link.
3. Hold `RET` down, letting buffer scroll up, and line starting with a link passes to top of window where point is. At which point `RET` may open the link several times in external browser.

## Versions of mu, mu4e/emacs, operating system etc.

```
mu (mail indexer/searcher) version 1.0
GNU Emacs 25.3.1 (x86_64-apple-darwin17.4.0, NS appkit-1561.20 Version 10.13.3 (Build 17D102)) of 2018-03-30
```

## Any other detail

Should `RET` not follow links, but rather `M-RET`? I think a similar decision was made here, for attachments: https://github.com/djcb/mu/commit/39147318748c4ba4b645612686aef3b33f07b15d.

I'm not sure if this issue comment is relevant, https://github.com/djcb/mu/pull/295#issuecomment-26823675.

**Comment by djcb** [comment]
Yeah `M-RET` would be good for "browse url".

_However_, this is in `shr`, so we can't change the default directly. Instead, I'd recommend something like in the comment; perhaps it's as simple as a single line, updating the mode-map for `shr`. If that works for you, please let us know, and we can add it as an example to the documentation.

**Comment by bradyt** [comment]
I just gave it a quick try, I might use something like the following as a hot fix here.

```elisp
(define-key shr-map (kbd "RET") 'mu4e-scroll-up)
(define-key shr-map (kbd "M-RET") 'shr-browse-url)
```

I was looking at the pseudocode from `(info "(elisp) Searching Keymaps")`, and tried `M-x describe-text-properties` with cursor over a link in `*mu4e-view*`, I think that's what we want to rebind. I tried `(make-local-variable 'shr-map)`, before doing the above, it seemed to work, and `RET` seemed to still work in `eww`, but maybe I did not give it a long enough try.
